packages/liveblocks-react/src/types.ts

```diff
@@ -83,6 +80,122 @@ export type UserState<T> =
   | UserStateError
   | UserStateSuccess<T>;
 
+export type PartialNullable<T> = {
+  [P in keyof T]?: T[P] | null | undefined;
+};
+
+export type CreateThreadOptions<TMetadata extends BaseMetadata> = [
+  TMetadata,
+] extends [never]
+  ? {
+      body: CommentBody;
+    }
+  : { body: CommentBody; metadata: TMetadata };
+
+export type EditThreadMetadataOptions<TMetadata extends BaseMetadata> = [
+  TMetadata,
+] extends [never]
+  ? {
+      threadId: string;
+    }
+  : { threadId: string; metadata: Resolve<PartialNullable<TMetadata>> };
+
+export type CreateCommentOptions = {
+  threadId: string;
+  body: CommentBody;
+};
+
+export type EditCommentOptions = {
+  threadId: string;
+  commentId: string;
+  body: CommentBody;
+};
+
+export type DeleteCommentOptions = {
+  threadId: string;
+  commentId: string;
+};
+
+export type CommentReactionOptions = {
+  threadId: string;
+  commentId: string;
+  emoji: string;
+};
+
+export type ThreadsStateLoading = {
+  isLoading: true;
+  threads?: never;
+  error?: never;
+};
+
+export type ThreadsStateResolved<TThreadMetadata extends BaseMetadata> = {
+  isLoading: false;
+  threads: ThreadData<TThreadMetadata>[];
+  error?: Error;
+};
+
+export type ThreadsStateSuccess<TThreadMetadata extends BaseMetadata> = {
+  isLoading: false;
+  threads: ThreadData<TThreadMetadata>[];
+  error?: never;
+};
+
+export type ThreadsState<TThreadMetadata extends BaseMetadata> =
+  | ThreadsStateLoading
+  | ThreadsStateResolved<TThreadMetadata>;
+
+export type InboxNotificationsStateLoading = {
+  isLoading: true;
+  inboxNotifications?: never;
+  error?: never;
+  loadMore: () => void;
```

Comment:
Should `loadMore` be returned when the inbox notifications query is still loading?

```tsx
export type InboxNotificationsStateLoading = {
  isLoading: true;
  inboxNotifications?: never;
  error?: never;
  loadMore: () => void;
};
```

Reply:
@nimeshnayaju It should always be returned yes, to avoid having to conditionally call it. But we can decide to make it a noop while the query is still loading, or keep the currently loading query and start the next one already?